libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -104,111 +100,201 @@ - (void)dealloc {
-    [defaults synchronize];
-    return loginHost;
+    return [SFSDKAuthPreferences sharedPreferences].loginHost;
 }
 
-#pragma mark - persistent properties
 - (NSSet *)scopes
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    NSArray *scopesArray = [defs objectForKey:kOAuthScopesKey] ?: [NSArray array];
-    return [NSSet setWithArray:scopesArray];
+    return [SFSDKAuthPreferences sharedPreferences].scopes;
 }
 
 - (void)setScopes:(NSSet *)newScopes
 {
-    NSArray *scopesArray = [newScopes allObjects];
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:scopesArray forKey:kOAuthScopesKey];
-    [defs synchronize];
+    [SFSDKAuthPreferences sharedPreferences].scopes = newScopes;
 }
 
 - (NSString *)oauthCompletionUrl
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    NSString *redirectUri = [defs objectForKey:kOAuthRedirectUriKey];
-    return redirectUri;
+    return [SFSDKAuthPreferences sharedPreferences].oauthCompletionUrl;
 }
 
 - (void)setOauthCompletionUrl:(NSString *)newRedirectUri
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:newRedirectUri forKey:kOAuthRedirectUriKey];
-    [defs synchronize];
+    [SFSDKAuthPreferences sharedPreferences].oauthCompletionUrl = newRedirectUri;
 }
 
 - (NSString *)oauthClientId
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    NSString *clientId = [defs objectForKey:kOAuthClientIdKey];
-    return clientId;
+    return [SFSDKAuthPreferences sharedPreferences].oauthClientId;
 }
 
 - (void)setOauthClientId:(NSString *)newClientId
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:newClientId forKey:kOAuthClientIdKey];
-    [defs synchronize];
+   [SFSDKAuthPreferences sharedPreferences].oauthClientId = newClientId;
 }
 
+#pragma  mark - login & logout
+
+- (BOOL)handleAdvancedAuthenticationResponse:(NSURL *)appUrlResponse {
+    [SFSDKCoreLogger i:[self class] format:@"handleAdvancedAuthenticationResponse %@",[appUrlResponse description]];
+    
+    SFSDKOAuthClient *client = [SFSDKOAuthClient nativeBrowserAuthInstance:nil];
+    return [client handleURLAuthenticationResponse:appUrlResponse];
+}
+
+- (BOOL)loginWithCompletion:(SFUserAccountManagerSuccessCallbackBlock)completionBlock failure:(SFUserAccountManagerFailureCallbackBlock)failureBlock {
+    SFOAuthCredentials *clientCredentials = [self retrieveClientCredentials];
+    return [self refreshCredentials:clientCredentials completion:completionBlock failure:failureBlock];
+}
+
+- (BOOL)refreshCredentials:(SFOAuthCredentials *)credentials completion:(SFUserAccountManagerSuccessCallbackBlock)completionBlock failure:(SFUserAccountManagerFailureCallbackBlock)failureBlock {
+    
+    SFSDKOAuthClient *client = [SFSDKOAuthClient clientWithContextBlock:^(SFSDKOAuthClientConfig *config,SFSDKMutableOAuthClientContext *context) {
+        context.credentials = credentials;
```

Comment:
Given that this is the only item used in this builder pattern for `SFSDKOAuthClientContext`, I'd prefer that the constructor simply be amended to add this as a parameter.  E.g. this call would become:
```obj-c
SFSDKOAuthClient *client = [SFSDKOAuthClient clientCredentials:credentials configBlock:^(SFSDKOAuthClientConfig *config) {
    config.advancedAuthConfiguration = self.advancedAuthConfiguration;
    config.delegate = self;
    config.webViewDelegate = self;
    config.safariViewDelegate = self;
    config.successCallbackBlock = completionBlock;
    config.failureCallbackBlock = failureBlock;
}];
```
It would obviate the need to change `SFOAuthCoordinator` from its current paradigm of being configured up front with credentials—and indeed, `SFOAuthClient` should really just be a front-end to that class.